modules/compute/function-app/main.bicep

```diff
@@ -0,0 +1,314 @@
+  properties: {
+    Application_Type: appInsightsType
+    WorkspaceResourceId: workspaceResourceId
+    publicNetworkAccessForIngestion: publicNetworkAccessForIngestion
+    publicNetworkAccessForQuery: publicNetworkAccessForQuery
+  }
+  tags: tags
+}
+
+
+@description('''The app or function app resource.
+Note: This is not actual Azure Function App this will be container for storing multiple functions.''')
+resource sites 'Microsoft.Web/sites@2022-03-01' = {
+  name: name
+  location: location
+  tags: tags
+  kind: kind
+  identity: {
+    type: identityType
+    userAssignedIdentities: (identityType == 'UserAssigned' || identityType == 'SystemAssigned, UserAssigned') ? {
+      '${userAssignedIdentityId}': {}
+    } : null
+  }
+  properties: {
+    serverFarmId: serverfarms.id
+    httpsOnly: httpsOnly
+    hostingEnvironmentProfile: !empty(appServiceEnvironmentId) ? {
+      id: appServiceEnvironmentId
+    } : null
+    clientAffinityEnabled: clientAffinityEnabled
+  }
+}
+
+
+@description('Appsettings/config for the sites (app or functionapp).')
+resource config 'Microsoft.Web/sites/config@2021-02-01' = {
+  parent: sites
+  name: 'appsettings'
+  properties: union({
+      AzureWebJobsStorage: !empty(storageAccount.id) ? 'DefaultEndpointsProtocol=https;AccountName=${storageAccount.name};AccountKey=${storageAccount.listKeys().keys[0].value};' : any(null)
+      AzureWebJobsDashboard: !empty(storageAccount.id) ? 'DefaultEndpointsProtocol=https;AccountName=${storageAccount.name};AccountKey=${storageAccount.listKeys().keys[0].value};' : any(null)
+      WEBSITE_CONTENTSHARE: name
+      WEBSITE_CONTENTAZUREFILECONNECTIONSTRING: !empty(storageAccount.id) ? 'DefaultEndpointsProtocol=https;AccountName=${storageAccount.name};AccountKey=${storageAccount.listKeys().keys[0].value};' : any(null)
+      FUNCTIONS_EXTENSION_VERSION: functionsExtensionVersion
+      FUNCTIONS_WORKER_RUNTIME: sites.kind == 'functionapp' && !empty(functionsWorkerRuntime) ? functionsWorkerRuntime : any(null)
+      WEBSITE_NODE_DEFAULT_VERSION: sites.kind == 'functionapp' && functionsWorkerRuntime == 'node' && !empty(functionsDefaultNodeversion) ? functionsDefaultNodeversion : any(null)
+      APPINSIGHTS_INSTRUMENTATIONKEY: !empty(appInsights.id) && enableInsights ? appInsights.properties.InstrumentationKey : any(null)
+      APPLICATIONINSIGHTS_CONNECTION_STRING: !empty(appInsights.id) && enableInsights ? appInsights.properties.ConnectionString : any(null)
+    }, extraAppSettings)
+  dependsOn: enableVnetIntegration ? [ networkConfig ] : []
+}
+
+
+resource networkConfig 'Microsoft.Web/sites/networkConfig@2022-03-01' = if (enableVnetIntegration == true) {
+  parent: sites
+  name: 'virtualNetwork'
+  properties: {
+    subnetResourceId: subnetId
+  }
+}
+
+
+@description('The resources actual is function where code exits')
+@batchSize(1)
+resource azureFunction 'Microsoft.Web/sites/functions@2021-02-01' = [for function in functions: {
+  dependsOn: [
```

Comment:
Is this dependsOn necessary?

Reply:
Hosting plan is not required for Function app but it's required as dependency fro premium plan